### ISSUE TITLE
[RFE] kola: add a `pre-flight` check

### ISSUE THREAD

**Issue by tormath1** [flatcar/Flatcar#470]
## Current situation

we use [kola](https://github.com/kinvolk/mantle/tree/flatcar-master/kola) to run tests on the CI but it can also be used locally. Since the  tool owns various options and has several requirements it would be helpful to provide a `pre-flight` check in order to catch any mis-configuration on the host or in the options in order to prevent eventual failures.

## Impact

it could reduce some friction on the `kola` usage.

## Ideal future situation

something like:

```
$ kola run ...
requirements:
    - [*] ipv4 forwarding
    - [ ] iptables nat rules
    - [*] kolet binary
```

## Implementation options

TBD

## Additional information

TBD

**Comment by jepio** [comment]
The iptables nat rules get autocreated by kola - i think it would be acceptable for kola to enable ipv4 forwarding as well.

**Comment by wrl** [comment]
@jepio I could go either way on that. enabling ipv4 forwarding is easy enough that it may make sense to have the user do it themselves, especially if we also need pre-flight checks for other dependencies.

**Comment by tormath1** [comment]
@jepio @wrl that's right - iptables are generated on the fly as a part of internet access (see this PR: https://github.com/kinvolk/mantle/pull/167) but ipv4 is not automagically enabled, it's up to the user to handle it. There are also things like `kolet` who needs to be in specific location (see: https://github.com/kinvolk/mantle/blob/edd19b0404eb563b73bd5657c865af9339b23679/kola/harness.go#L583-L589)

To be consistent, there is two options to setup the correct environment:

- rely on kola
- rely on the user / documentation 

If we rely :100: on kola, we are not sure we could provide an idempotent behavior or revert the system to its initial state.

**Comment by pothos** [comment]
Reverting IP forwarding to being disabled is also not done by systemd: https://www.freedesktop.org/software/systemd/man/systemd.network.html#IPForward=
I think it's fine to turn it on and leave it on.

**Comment by wrl** [comment]
agreed with @tormath1. in my opinion, it's fine to rely on the user/documentation for some of the setup (we certainly shouldn't expect kola to install dnsmasq or qemu), but where we *do* rely on the user/documentation, a pre-flight check to ensure that the requirements are met is prudent.

I think whether IPv4 forwarding is handled by the user/documentation or by kola directly could really go either way.

**Comment by tormath1** [comment]
a PR has been opened to provide a bit more of documentation around the kola tests: https://github.com/kinvolk/mantle/pull/223 :)

**Comment by tormath1** [comment]
I think the whole thing is correctly documented now and we have not been aware of setup issues in the last years. Let's close this. :)
